### PR TITLE
{xmake-core-sv,tbox,xmake}: init

### LINKED ISSUE #212231: Add xmake
**Project description**

A lightweight cross-platform build utility based on Lua

**Metadata**

* homepage URL: https://xmake.io
* source URL: https://github.com/xmake-io/xmake
* license: asl20
* platforms: unix

### PR DESCRIPTION
**Pull request by wineee**
###### Description of changes

Superseded https://github.com/NixOS/nixpkgs/pull/177881
close https://github.com/NixOS/nixpkgs/issues/212231

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/xmake/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    hash = "sha256-m0LYY0gz9IhbBbiUKd1gBE3KmSMvYJYyC42Ff7M9Ku8=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    luaPackages.lua
+    luaPackages.cjson
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/xmake/default.nix:28]:
```suggestion
    lua
    lua.pkgs.cjson
```

File: pkgs/development/tools/build-managers/xmake/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    xmake-core-sv
+  ];
+
+  configureFlags = [ "--external=y" ];
+
+  meta = with lib; {
+    description = "A cross-platform build utility based on Lua";
+    homepage = "https://xmake.io";
+    license = licenses.asl20;
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/build-managers/xmake/default.nix:42]:
```suggestion
    platforms = lua.meta.platforms;
```

File: pkgs/development/libraries/tbox/libtbox.pc.in [hunk 3]
```diff
@@ -0,0 +1,10 @@
+prefix=@out@
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tbox/libtbox.pc.in:1]: Please try to upstream it

Comment (wineee) [reply pkgs/development/libraries/tbox/libtbox.pc.in:1]: Upstream provides pkgconfig files via `xrepo`，use  `xmake install -o xxx` to get .pc file (and the binary）


We use [xmake.sh](https://github.com/xmake-io/xmake.sh) to build tbox, currently does not support exporting `.pc` file

**Pull request merged**